src/server.js

```diff
@@ -23,12 +29,111 @@ const sendUserError = (err, res) => {
   }
 };
 
+const handleLogin = (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username) {
+    sendUserError('No username', res);
+    return;
+  }
+  if (!password) {
+    sendUserError('No password', res);
+    return;
+  }
+  User.find({ username })
+    .then(user => {
+      req.hashedPassword = user[0].passwordHash;
+      next();
+    })
+    .catch(err => {
+      sendUserError(err, res);
+    });
+};
+
+const checkLoggedIn = (req, res, next) => {
+  if (req.session.loggedIn) {
+    User.findOne({ username: req.session.username })
+      .then(user => {
+        req.user = user;
+        next();
+      })
+      .catch(err => {
+        sendUserError(err, res);
+      });
+  } else {
+    sendUserError('You must log in first', res);
+  }
+};
+
```

Comment:
To keep your error messages consistent you may want to choose either `log in` or `login`.